libkbfs/bare_root_metadata.go

```diff
@@ -516,10 +516,37 @@ func (md *BareRootMetadata) getTLFKeyBundles(keyGen KeyGen) (
 	return &md.WKeys[i], &md.RKeys[i], nil
 }
 
+// GetDeviceKIDs returns the KIDs (of CryptPublicKeys) for all known
+// devices for the given user at the given key generation, if any.
+// Returns an error if the TLF is public, or if the given key
+// generation is invalid.
+func (md *BareRootMetadata) GetDeviceKIDs(
+	keyGen KeyGen, user keybase1.UID) ([]keybase1.KID, error) {
+	wkb, rkb, err := md.getTLFKeyBundles(keyGen)
+	if err != nil {
+		return nil, err
+	}
+
+	dkim := wkb.WKeys[user]
+	if len(dkim) == 0 {
+		dkim = rkb.RKeys[user]
+	}
+
+	var kids []keybase1.KID
+	for kid := range dkim {
+		kids = append(kids, kid)
+	}
+
+	return kids, nil
```

Comment:
It already does, since `kids` starts off as empty. But may as well size it to `len(dkim)` and return early if `dkim` is empty, so did that.